### PR TITLE
Remove deprecated property `is_slurm_managing_tasks` from accelerator connector

### PR DESCRIPTION
**Pull request by SauravMaheshkar**
## What does this PR do?

Fixes a part of  #10312

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Yesss !! ☕️

### CODE REVIEW

**Comment by SauravMaheshkar** [comment]
**Question for the maintainers: 🧐🧐🧐**

Should I remove the following as well ??

https://github.com/PyTorchLightning/pytorch-lightning/blob/ce3e63262a7247c1111375b28c90bdd83b42bf3f/pytorch_lightning/trainer/connectors/accelerator_connector.py#L1016-L1042

File: pytorch_lightning/trainer/connectors/accelerator_connector.py [hunk 1]
```diff
@@ -993,20 +989,6 @@ def update_device_type_if_training_type_plugin_passed(self) -> None:
...
-        )
-        return self._is_slurm_managing_tasks
-
-    @is_slurm_managing_tasks.setter
-    def is_slurm_managing_tasks(self, value: bool) -> bool:
-        rank_zero_deprecation(
-            "`AcceleratorConnector.is_slurm_managing_tasks` was deprecated in v1.5 and will be removed in v1.6."
-        )
-        self._is_slurm_managing_tasks = value
-
```
Comment (awaelchli) [inline pytorch_lightning/trainer/connectors/accelerator_connector.py:1009]: You only need to remove this code, the public methods. 
The protected attribute needs to stay and will be taken care of in my PR here #10102
It is part of a larger simplification and improvement.

File: pytorch_lightning/trainer/connectors/accelerator_connector.py [hunk 2]
```diff
@@ -810,8 +808,6 @@ def select_accelerator(self) -> Accelerator:
     def select_cluster_environment(self) -> ClusterEnvironment:
         if self._cluster_environment is not None:
             return self._cluster_environment
-        if self._is_slurm_managing_tasks:
-            env = SLURMEnvironment()
```
Comment (awaelchli) [inline pytorch_lightning/trainer/connectors/accelerator_connector.py:814]: This needs to stay :)

**Review by awaelchli — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -65,9 +65,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 - Removed deprecated `Trainer.fit(train_dataloader=)`, `Trainer.validate(val_dataloaders=)`, and `Trainer.test(test_dataloader=)` ([#10325](https://github.com/PyTorchLightning/pytorch-lightning/pull/10325))
 
-
-
--
+- Remove deprecated property _slurm_managing_tasks from accelerator connector ([#10353](https://github.com/PyTorchLightning/pytorch-lightning/pull/10353))
```
Comment (awaelchli) [inline CHANGELOG.md:68]:
```suggestion
- Remove deprecated property slurm_managing_tasks from accelerator connector ([#10353](https://github.com/PyTorchLightning/pytorch-lightning/pull/10353))
```

File: pytorch_lightning/trainer/connectors/accelerator_connector.py (hunk 1 above)
Comment (SauravMaheshkar) [reply pytorch_lightning/trainer/connectors/accelerator_connector.py:1009]: Done, wasn't aware that I just had to remove these

Comment (awaelchli) [reply pytorch_lightning/trainer/connectors/accelerator_connector.py:1009]: Yep, I fully understand

Thanks a lot for taking this <3

**Comment by SauravMaheshkar** [comment]
Just to clarify @awaelchli, I should remove the tests as well right ?? AFAIK they're the reason for the failing tests.

**Comment by awaelchli** [comment]
Yes @SauravMaheshkar 
See my three latest commits. I helped to remove the unintended changes. The only thing you needed to remove were the public attributes for `is_slurm_managing_tasks` and the corresponding tests. The tests were in place to ensure that the properties are kept and appropriate deprecation message is shown. The PR should now be ready to merge.

**Review by awaelchli — APPROVED** [review]
Great! 
Would you like to do a next PR for the method `configure_slurm_ddp`?

**Comment by SauravMaheshkar** [comment]
Thanks for the clarification, more than happy to take `configure_slurm_dpp` next 😁.

**Comment by SauravMaheshkar** [comment]
Taking up `configure_slurm_ddp` in #10370

**Review by justusschock — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

**Pull request merged**